### PR TITLE
Add SR-IOV Device Plugin Support

### PR DESCRIPTION
**Pull request by wizhaoredhat**
Some of the design decisions to add the SR-IOV device plugin daemonset are temporary.

### CODE REVIEW

**Comment by wizhaoredhat** [comment]
```
KUBEBUILDER_ASSETS="/home/wizhao/repo/github_ext/openshift/openshift-dpu-operator/bin/k8s/1.27.1-linux-amd64" go test ./... -coverprofile cover.out
?       github.com/openshift/dpu-operator/cmd   [no test files]
?       github.com/openshift/dpu-operator/daemon/plugin [no test files]
?       github.com/openshift/dpu-operator/dpu-cni       [no test files]
?       github.com/openshift/dpu-operator/dpu-cni/example       [no test files]
?       github.com/openshift/dpu-operator/dpu-cni/pkgs/cni      [no test files]
?       github.com/openshift/dpu-operator/dpu-cni/pkgs/cnitypes [no test files]
?       github.com/openshift/dpu-operator/dpu-cni/pkgs/sriov    [no test files]
?       github.com/openshift/dpu-operator/dpu-cni/pkgs/sriovconfig      [no test files]
?       github.com/openshift/dpu-operator/dpu-cni/pkgs/sriovtypes       [no test files]
?       github.com/openshift/dpu-operator/dpu-cni/pkgs/sriovutils       [no test files]
ok      github.com/openshift/dpu-operator/daemon        0.023s  coverage: 43.7% of statements
ok      github.com/openshift/dpu-operator/dpu-cni/pkgs/cnihelper        0.005s  coverage: 44.4% of statements
ok      github.com/openshift/dpu-operator/dpu-cni/pkgs/cnilogging       0.008s  coverage: 100.0% of statements
ok      github.com/openshift/dpu-operator/dpu-cni/pkgs/cniserver        0.010s  coverage: 54.0% of statements
ok      github.com/openshift/dpu-operator/internal/controller   26.253s coverage: 70.0% of statements
```

File: internal/controller/dpuoperatorconfig_controller.go [hunk 1]
```diff
@@ -84,15 +88,19 @@ func getImagePullPolicy() string {
...
+}
+
 func (r *DpuOperatorConfigReconciler) ensureDpuDeamonSetRunning(ctx context.Context, cfg *configv1.DpuOperatorConfig) error {
 	var err error
 
 	logger := log.FromContext(ctx)
 	data := render.MakeRenderData()
 	// All the CRs will be in the same namespace as the operator config
-	data.Data["Namespace"] = cfg.Namespace
+	setCommonData(&data, cfg)
```
Comment (bn222) [inline internal/controller/dpuoperatorconfig_controller.go:102]: nice

File: bindata/sriov-device-plugin/04.sriov-device-plugin_ds.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    metadata:
+      labels:
+        name: sriov-device-plugin
+        component: network
+        type: infra
+        openshift.io/component: network
+    spec:
+      hostNetwork: true
+      nodeSelector:
+      # TODO: We need to define how we want to select nodes to run the device plugin. Perhaps "dpu: "true""
```
Comment (bn222) [inline bindata/sriov-device-plugin/04.sriov-device-plugin_ds.yaml:24]: Yes, this is what I was missing. Can you hardcode the label dpu=true here?

**Comment by bn222** [comment]
only 1 comment. Thanks.

**Pull request merged**